common/script/ops/buy.js

```diff
@@ -3,117 +3,133 @@ import i18n from '../i18n';
 import _ from 'lodash';
 import count from '../count';
 import splitWhitespace from '../libs/splitWhitespace';
+import {
+  BadRequest,
+  NotAuthorized,
+  NotFound,
+} from '../libs/errors';
+import predictableRandom from '../fns/predictableRandom';
```

Comment:
The reason it's still not working is because you're referencing the object this way instead of `shared.fns.predictableRandom` or `fns.predictableRandom`. Sinon is wrapping the the parent object and replacing the specific function. In this case you're calling the function directly, so it's not being wrapped. 

If these were imported from a `fns` object, it should work.